src/Nethermind/Nethermind.Network/TrustedNodes/TrustedNodesManager.cs

```diff
@@ -0,0 +1,182 @@
+// SPDX-FileCopyrightText: 2024 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net.Sockets;
+using System.Runtime.CompilerServices;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using Nethermind.Config;
+using Nethermind.Core.Crypto;
+using Nethermind.Logging;
+using Nethermind.Stats.Model;
+
+namespace Nethermind.Network
+{
+    public class TrustedNodesManager : ITrustedNodesManager
+    {
+        private ConcurrentDictionary<PublicKey, NetworkNode> _nodes = new();
+        private readonly string _trustedNodesPath;
+        private readonly ILogger _logger;
+
+        public TrustedNodesManager(string trustedNodesPath, ILogManager logManager)
+        {
+            _trustedNodesPath = trustedNodesPath.GetApplicationResourcePath();
+            _logger = logManager.GetClassLogger();
+        }
+
+        public IEnumerable<NetworkNode> Nodes => _nodes.Values;
+
+        public async Task InitAsync()
+        {
+            if (!File.Exists(_trustedNodesPath))
+            {
+                if (_logger.IsDebug)
+                {
+                    _logger.Debug($"Trusted nodes file not found at: {_trustedNodesPath}");
+                }
```

Comment:
generally we do those `if` for log levels as one-liners, to avoid cluttering code

```suggestion
                if (_logger.IsDebug) _logger.Debug($"Trusted nodes file not found at: {_trustedNodesPath}");
```